agent/engine/docker_task_engine.go

```diff
@@ -572,10 +583,16 @@ func (engine *DockerTaskEngine) concurrentPull(task *api.Task, container *api.Co
 	defer seelog.Debugf("Released ImagePullDeleteLock after pulling image - %s. Task: %v", container.Image, task)
 	defer ImagePullDeleteLock.RUnlock()
 
-	pullStart := time.Now()
+	// Record the task pull_started_at timestamp
+	pullStart := engine.time().Now()
 	defer func(startTime time.Time) {
 		seelog.Infof("Finished pulling container %v in %s. Task: %v", container.Image, time.Since(startTime).String(), task)
 	}(pullStart)
+	ok := task.SetPullStartedAt(pullStart)
+	if ok {
+		seelog.Infof("Task started pull, task %s, time: %s", task.String(), pullStart)
```

Comment:
Let's change this message to be more clear about timestamp being recorded as well: `"Recording timestamp for starting image pull, task: ..."`